src/snowflake/cli/api/commands/decorators.py

```diff
@@ -67,7 +68,9 @@ def _decorator(func: Callable):
                     "project_definition",
                     inspect.Parameter.KEYWORD_ONLY,
                     annotation=Optional[str],
-                    default=project_type_option(project_name),
+                    default=project_type_option(project_name)
+                    if project_name
+                    else project_definition_option(),
```

Comment:
Is this a short-term fix? It seems to me like we should just make `project_type_option` take an `Optional[str]` as well, I'm not sure what having two paths gets us with such similar names / purposes.

Reply:
Imho the `project_name` option does not create enough value to have it. My take would be that this decorator should provide full project definition. If command requires a part of it (for example streamlit) then the command should access it.